clickhouse-client/src/main/java/com/clickhouse/client/ClickHouseException.java

```diff
@@ -119,10 +118,7 @@ public static ClickHouseException forCancellation(Exception e, ClickHouseNode se
     public static boolean isConnectTimedOut(Throwable t) {
         if (t instanceof SocketTimeoutException || t instanceof TimeoutException) {
             String msg = t.getMessage();
-            if (msg != null && msg.length() >= MSG_CONNECT_TIMED_OUT.length()) {
-                msg = msg.substring(0, MSG_CONNECT_TIMED_OUT.length()).toLowerCase(Locale.ROOT);
-            }
-            return MSG_CONNECT_TIMED_OUT.equals(msg);
+            return msg != null && msg.toLowerCase().contains(MSG_CONNECT_TIMED_OUT);
```

Comment:
Probably this logic for error detection should be more client-specific. I have some ideas how it can be implemented, but it requires more efforts for  implementation. 
As a fast fix this solution looks good from my point of view.

Reply:
Parsing error message is too fragile. We should use error codes or exception classes. 
The old code misses some other cases. For example, if we were waiting for a task and failed with TimeoutException it will not be a connection timeout, but in some cases it means that network is disrupted and we need to switch server.